0867725... Make sort-options experimental.  Part II.  Fixes crasher n-44.patch.  Fixes crasher: n#442036.
7dcddcb... Make sort-options experimental.  Part I.
c7bb736... Add --enable-layout-experimental configure switch.  Part II.  Typo in configure.in, add set_soenv entry.
fea9cd2... Add --enable-layout-experimental configure switch.  Default off.

diff --git config_office/configure.in config_office/configure.in
index 531db1e..00b7ab3 100644
--- config_office/configure.in
+++ config_office/configure.in
@@ -239,6 +239,10 @@ AC_ARG_ENABLE(static-gtk,
 AC_ARG_ENABLE(layout,
 [  --disable-layout        Disable the compilation and use of layout dialogs
 ],,)
+AC_ARG_ENABLE(layout_experimental,
+[  --enable-layout-experimental  Enable the compilation and use of layout dialogs
+                               tagged experimental.
+],,)
 AC_ARG_ENABLE(build-mozilla,
 [  --disable-build-mozilla Use this option if you do not want to build the
                           mozilla components from the mozilla source code but
@@ -6471,6 +6475,16 @@ else
 fi
 AC_SUBST(ENABLE_LAYOUT)
 
+AC_MSG_CHECKING([whether to use experimental layout dialogs])
+if test -n "$enable_layout_experimental" && test "$enable_layout_experimental" == "yes"; then
+   ENABLE_LAYOUT_EXPERIMENTAL="TRUE"
+   AC_MSG_RESULT([yes])
+else
+   ENABLE_LAYOUT_EXPERIMENTAL="FALSE"
+   AC_MSG_RESULT([no])
+fi
+AC_SUBST(ENABLE_LAYOUT_EXPERIMENTAL)
+
 # ===================================================================
 # De- or increase default verbosity of build process
 # ===================================================================
diff --git a/config_office/set_soenv.in b/config_office/set_soenv.in
index c6bd1de..bf515bc 100644
--- config_office/set_soenv.in
+++ config_office/set_soenv.in
@@ -1982,6 +1982,7 @@ ToFile( "ENABLE_RANDR",      "@ENABLE_RANDR@",      "e" );
 ToFile( "ENABLE_FONTCONFIG", "@ENABLE_FONTCONFIG@", "e" );
 ToFile( "ENABLE_DIRECTX",    "@ENABLE_DIRECTX@",    "e" );
 ToFile( "ENABLE_LAYOUT",     "@ENABLE_LAYOUT@",     "e" );
+ToFile( "ENABLE_LAYOUT_EXPERIMENTAL", "@ENABLE_LAYOUT_EXPERIMENTAL@", "e" );
 ToFile( "ENABLE_PCH",        "@ENABLE_PCH@",       "e" );
 ToFile( "VC_STANDARD",       "@VC_STANDARD@",      "e" );
 ToFile( "WITH_GPC",          "@WITH_GPC@",         "e" );
diff --git sc/source/ui/dbgui/tpsort.cxx sc/source/ui/dbgui/tpsort.cxx
index ead9819..d17a08a 100644
--- sc/source/ui/dbgui/tpsort.cxx
+++ sc/source/ui/dbgui/tpsort.cxx
@@ -536,6 +536,7 @@ IMPL_LINK( ScTabPageSortFields, SelectHdl, ListBox *, pLb )
 // Sortieroptionen-Tabpage:
 //========================================================================
 
+#if ENABLE_LAYOUT_EXPERIMENTAL
 #include <layout/layout-pre.hxx>
 
 #if ENABLE_LAYOUT
@@ -545,6 +546,8 @@ IMPL_LINK( ScTabPageSortFields, SelectHdl, ListBox *, pLb )
 #define SfxTabPage( parent, id, args ) TabPage( parent, "sort-options.xml", id, 0, &args )
 #endif /* ENABLE_LAYOUT */
 
+#endif /* ENABLE_LAYOUT_EXPERIMENTAL */
+
 ScTabPageSortOptions::ScTabPageSortOptions( Window* 			pParent,
 											const SfxItemSet&	rArgSet )
 
@@ -571,10 +574,10 @@ ScTabPageSortOptions::ScTabPageSortOptions( Window* 			pParent,
 // 		aFtAreaLabel	( this, ScResId( FT_AREA_LABEL ) ),
 //		aFtArea 		( this, ScResId( FT_AREA ) ),
 		//
-#if ENABLE_LAYOUT
+#if ENABLE_LAYOUT_EXPERIMENTAL
 #undef ScResId
 #define ScResId(x) this, #x
-#endif /* ENABLE_LAYOUT */
+#endif /* ENABLE_LAYOUT_EXPERIMENTAL */
 		aStrRowLabel	( ScResId( STR_ROW_LABEL ) ),
 		aStrColLabel	( ScResId( STR_COL_LABEL ) ),
 		aStrUndefined	( ScResId( SCSTR_UNDEFINED ) ),
@@ -715,10 +718,10 @@ USHORT* __EXPORT ScTabPageSortOptions::GetRanges()
 
 // -----------------------------------------------------------------------
 
-#if ENABLE_LAYOUT
+#if ENABLE_LAYOUT_EXPERIMENTAL
 #undef SfxTabPage
 #define SfxTabPage TabPage
-#endif /* ENABLE_LAYOUT */
+#endif /* ENABLE_LAYOUT_EXPERIMENTAL */
 SfxTabPage* __EXPORT ScTabPageSortOptions::Create(
 											Window*				pParent,
 											const SfxItemSet&	rArgSet )
@@ -895,11 +898,11 @@ int __EXPORT ScTabPageSortOptions::DeactivatePage( SfxItemSet* pSetP )
 
 		if ( !bPosInputOk )
 		{
-#if !ENABLE_LAYOUT
+#if !ENABLE_LAYOUT_EXPERIMENTAL
 			ErrorBox( this, WinBits( WB_OK | WB_DEF_OK ),
 					 ScGlobal::GetRscString( STR_INVALID_TABREF )
 					).Execute();
-#endif /* ENABLE_LAYOUT */
+#endif /* ENABLE_LAYOUT_EXPERIMENTAL */
 			aEdOutPos.GrabFocus();
 			aEdOutPos.SetSelection( Selection( 0, SELECTION_MAX ) );
 			theOutPos.Set(0,0,0);
diff --git sc/source/ui/inc/tpsort.hxx sc/source/ui/inc/tpsort.hxx
index ccf5db0..c1fe02c 100644
--- sc/source/ui/inc/tpsort.hxx
+++ sc/source/ui/inc/tpsort.hxx
@@ -132,8 +132,12 @@ class ScRangeData;
 class CollatorRessource;
 class CollatorWrapper;
 
+#if ENABLE_LAYOUT_EXPERIMENTAL
 #include <layout/layout.hxx>
 #include <layout/layout-pre.hxx>
+#else /* !ENABLE_LAYOUT_EXPERIMENTAL */
+#define LocalizedString String
+#endif /* !ENABLE_LAYOUT_EXPERIMENTAL */
 
 class ScTabPageSortOptions : public SfxTabPage
 {
@@ -214,7 +218,9 @@ private:
 #endif
 };
 
+#if ENABLE_LAYOUT_EXPERIMENTAL
 #include <layout/layout-post.hxx>
+#endif /* ENABLE_LAYOUT_EXPERIMENTAL */
 
 #endif // SC_TPSORT_HXX
 
diff --git solenv/inc/settings.mk solenv/inc/settings.mk
index 60e9bca..9ac66a8 100644
--- solenv/inc/settings.mk
+++ solenv/inc/settings.mk
@@ -1147,6 +1147,12 @@ CDEFS+=-DENABLE_LAYOUT=1
 CFLAGS+=-DENABLE_LAYOUT=0
 .ENDIF # ENABLE_LAYOUT != TRUE
 
+.IF "$(ENABLE_LAYOUT_EXPERIMENTAL)" == "TRUE"
+CDEFS+=-DENABLE_LAYOUT_EXPERIMENTAL=1
+.ELSE # ENABLE_LAYOUT_EXPERIMENTAL != TRUE
+CFLAGS+=-DENABLE_LAYOUT_EXPERIMENTAL=0
+.ENDIF # ENABLE_LAYOUT_EXPERIMENTAL != TRUE
+
 # compose flags and defines for GUI
 .IF "$(TARGETTYPE)"=="GUI"
 CDEFS+= $(CDEFSGUI)
